analyses/mutect2-vs-strelka2/01-set-up.Rmd

```diff
@@ -0,0 +1,476 @@
+    # Make a new column that specifies what maf object the data is from
+    dplyr::mutate(dataset = dplyr::recode(dataset,
+      `TRUE` = maf1_name,
+      `FALSE` = maf2_name,
+    )) %>%
+    # Gets rid of the ".x" and ".y" specifications of the column names
+    dplyr::mutate(variable = gsub(".x$|.y$", "", variable)) %>%
+    # Spreads the data based on the new dataset variable
+    tidyr::spread("dataset", "value")
+
+  # Get correlations by each type of variant classification
+  cors <- combine_df %>%
+    dplyr::group_by(variable) %>%
+    dplyr::summarize(cor = cor(
+      # This `eval(parse(text =` piece is so that `cor` will correlate the
+      # numeric vectors not the character strings themselves
+      eval(parse(text = maf1_name)),
+      eval(parse(text = maf2_name)),
+      method = cor_method
+    ))
+
+  # Add the r values to the data labels
+  combine_df <- combine_df %>%
+    dplyr::mutate(variable_w_cor = paste(
+      variable,
+      "r =",
+      round(cors$cor, 3)
+    ))
+
+  # Plot this as a facet_wrapped scatterplot with the r values in the label
+  ggplot2::ggplot(combine_df, ggplot2::aes(
+    x = eval(parse(text = maf1_name)),
+    y = eval(parse(text = maf2_name))
+  )) +
+    ggplot2::geom_hex(bins = 10) +
+    ggplot2::facet_wrap(~variable_w_cor, scales = "free") +
+    ggplot2::xlab(maf1_name) +
+    ggplot2::ylab(maf2_name) +
+    ggplot2::theme_classic() +
+    ggplot2::theme(strip.text = ggplot2::element_text(size = 5))
+}
+```
+
+
+## Read in the metadata information
+
+Running `maftools::read.maf` takes a lot of computing power and time, so to 
+avoid having to run this for both datasets everytime we want to re-run this 
+notebook or the analyses in the other notebook, I've set this up to save the 
+`MAF` objects as `RDS` files.
+
+First let's establish the file paths.
+
+```{r}
+# File paths for the needed files for this analysis
+metadata_dir <- file.path(scratch_dir, "metadata_filtered_maf_samples.tsv")
+strelka2_dir <- file.path(scratch_dir, "strelka2.RDS")
+mutect2_dir <- file.path(scratch_dir, "mutect2.RDS")
+```
+
+## Read in the Strelka2 and Mutect2 data
+
+We will read in the data as `maftools` objects from an RDS file, unless 
+`maftools` has not been run on them yet.
+We will use `metadata` as the `clinicalData` for the `maftools` object. 
+ 
+Note: If you trying to run the initial set up step in a Docker container, it 
+will likely be out of memory killed, unless you have ~50GB you can allocate to 
+Docker. 
+
+```{r}
+# Get a vector of whether these exist
+files_needed <- file.exists(metadata_dir, strelka2_dir, mutect2_dir)
+
+if (all(files_needed)) {
+  # Read the ready-to-go files if these files exist
+  metadata <- metadata <- readr::read_tsv(metadata_dir)
+  strelka2 <- readRDS(strelka2_dir)
+  mutect2 <- readRDS(mutect2_dir)
+} else { # If any of the needed files don't exist, rerun this process:
+  # Only import the sample names
+  strelka2_samples <- data.table::fread(file.path(
```

Comment:
I think the indentation throughout here is off. I would expect the arguments to `data.table::fread` to be indented more.